checks/kubernetes_resource.go

```diff
@@ -277,3 +299,58 @@ func (c *KubernetesResourceChecker) validate(ctx *context.Context, check v1.Kube
 
 	return nil
 }
+
+// deleteResourceSync deletes the given kubernetes resource and waits for it to be deleted.
+func deleteResourceSync(ctx *context.Context, resource unstructured.Unstructured) error {
+	ctx.Logger.V(5).Infof("deleting resource (%s/%s/%s)", resource.GetKind(), resource.GetNamespace(), resource.GetName())
+
+	rc, err := ctx.Kommons().GetRestClient(resource)
+	if err != nil {
+		return fmt.Errorf("failed to get rest client config for (%s/%s/%s): %w", resource.GetKind(), resource.GetNamespace(), resource.GetName(), err)
+	}
+
+	watcher, err := rc.Watch(resource.GetNamespace(), resource.GetAPIVersion(), &metav1.ListOptions{})
```

Comment:
I don't think watches are meant to be used like this - they are fairly expensive on etcd. 

Simple polling should be better (but we should probably have a waitForDelete option, if the resource names can be made unique with a random value, then we don't need to wait)

For deletes the RestClient can be used - Make sure to specify GracePeriodSeconds: 0 and propagationPolicy: orphan on the DeleteOptions